sdk/python/kfp/dsl/component_factory.py

```diff
@@ -575,7 +599,8 @@ def create_component_from_func(
         output_component_file=output_component_file,
         base_image=base_image,
         packages_to_install=packages_to_install,
-        pip_index_urls=pip_index_urls)
+        pip_index_urls=pip_index_urls,
+        pip_trusted_hosts=pip_trusted_hosts)
```

Comment:
I noticed the tests additions mostly cover the cli build commands, it would be nice to see tests that also cover the dsl compilation portion

Reply:
The cli will call the code generation. Due that the new test is simple, I added it as well: You can double-check the method: test_with_packages_to_install_with_pip_index_url_and_trusted_host